types/dynamic_fee_tx.go

```diff
@@ -0,0 +1,330 @@
+package types
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/umbracle/fastrlp"
+)
+
+type DynamicFeeTx struct {
+	Nonce     uint64
+	GasTipCap *big.Int
+	GasFeeCap *big.Int
+	Gas       uint64
+	To        *Address
+	Value     *big.Int
+	Input     []byte
+	V, R, S   *big.Int
+	Hash      Hash
+	From      Address
+
+	ChainID    *big.Int
+	AccessList TxAccessList
+}
+
+func (tx *DynamicFeeTx) transactionType() TxType { return DynamicFeeTxType }
+func (tx *DynamicFeeTx) chainID() *big.Int       { return tx.ChainID }
+func (tx *DynamicFeeTx) input() []byte           { return tx.Input }
+func (tx *DynamicFeeTx) gas() uint64             { return tx.Gas }
+func (tx *DynamicFeeTx) gasPrice() *big.Int      { return tx.GasFeeCap }
+func (tx *DynamicFeeTx) gasTipCap() *big.Int     { return tx.GasTipCap }
+func (tx *DynamicFeeTx) gasFeeCap() *big.Int     { return tx.GasFeeCap }
+func (tx *DynamicFeeTx) value() *big.Int         { return tx.Value }
+func (tx *DynamicFeeTx) nonce() uint64           { return tx.Nonce }
+func (tx *DynamicFeeTx) to() *Address            { return tx.To }
+func (tx *DynamicFeeTx) from() Address           { return tx.From }
+
+func (tx *DynamicFeeTx) hash() Hash { return tx.Hash }
+
+func (tx *DynamicFeeTx) rawSignatureValues() (v, r, s *big.Int) {
+	return tx.V, tx.R, tx.S
+}
+
+func (tx *DynamicFeeTx) accessList() TxAccessList { return tx.AccessList }
+
+func (tx *DynamicFeeTx) setSignatureValues(v, r, s *big.Int) {
+	tx.V, tx.R, tx.S = v, r, s
+}
+
+func (tx *DynamicFeeTx) setFrom(addr Address) { tx.From = addr }
+
+func (tx *DynamicFeeTx) setGas(gas uint64) {
+	tx.Gas = gas
+}
+
+func (tx *DynamicFeeTx) setChainID(id *big.Int) {
+	tx.ChainID = id
+}
+
+func (tx *DynamicFeeTx) setGasPrice(gas *big.Int) {
+	tx.GasTipCap = gas
+}
+
+func (tx *DynamicFeeTx) setGasFeeCap(gas *big.Int) {
+	tx.GasFeeCap = gas
+}
+
+func (tx *DynamicFeeTx) setGasTipCap(gas *big.Int) {
+	tx.GasTipCap = gas
+}
+
+func (tx *DynamicFeeTx) setValue(value *big.Int) {
+	tx.Value = value
+}
+
+func (tx *DynamicFeeTx) setInput(input []byte) {
+	tx.Input = input
+}
+
+func (tx *DynamicFeeTx) setTo(address *Address) {
+	tx.To = address
+}
+
+func (tx *DynamicFeeTx) setNonce(nonce uint64) {
+	tx.Nonce = nonce
+}
+
+func (tx *DynamicFeeTx) setAccessList(accessList TxAccessList) {
+	tx.AccessList = accessList
+}
+
+func (tx *DynamicFeeTx) setHash(h Hash) { tx.Hash = h }
+
+// unmarshalRLPFrom unmarshals a Transaction in RLP format
+// Be careful! This function does not de-serialize tx type, it assumes that t.Type is already set
+// Hash calculation should also be done from the outside!
+// Use UnmarshalRLP in most cases
+func (tx *DynamicFeeTx) unmarshalRLPFrom(p *fastrlp.Parser, v *fastrlp.Value) error {
+	numOfElems := 12
+
+	elems, err := v.GetElems()
+	if err != nil {
+		return err
+	}
+
+	getElem := func() *fastrlp.Value {
+		val := elems[0]
+		elems = elems[1:]
+
+		return val
+	}
```

Comment:
Pls double check me on this, but seems at the first glance it should be feasible to refactor it in the following way. :pray:

Instead of inline `getElem`, I'd suggest introducing type alias and the following function to let's say `rlp_unmarshal.go` and reuse the `dequeueValue` wherever `getElem` is called.

```suggestion
type rlpValues []*fastrlp.Value

func (r rlpValues) dequeueValue() *fastrlp.Value {
	val := r[0]
	r = r[1:]

	return val
}
```